packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -250,7 +282,28 @@ function WorklistDataGrid(props) {
   useEffect(() => {
     setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [columns]);
+  }, [pinnedColumns, overflowColumns]);
+
+  // useEffect to calculate pinned column offsets
+  useEffect(() => {
+    if (pinnedColumns.length > 0) {
+      let cumulativeOffset = 0;
+      const offsetArray = [0];
+      const totalPinnedColumnsCount = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
+
+      // eslint-disable-next-line array-callback-return
+      dataGridColumns.slice(0, totalPinnedColumnsCount).map((pinnedCol) => {
```

Comment:
I would spell out pinnedColumn instead of using an abbreviation.

Reply:
Updated: [c366625](https://github.com/cerner/terra-framework/pull/1710/commits/c3666253e0347c3c0e39171d02a5c0008a162975)